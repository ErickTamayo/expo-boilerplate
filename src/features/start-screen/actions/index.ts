import { ADD_COUNTER } from '../constants'

export interface AddCounter {
  type: ADD_COUNTER
}

export const addCounter = (): AddCounter => ({
  type: ADD_COUNTER,
})
