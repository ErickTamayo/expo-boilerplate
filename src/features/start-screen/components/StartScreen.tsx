import * as React from 'react'
import { StyleSheet, Text, View, Button } from 'react-native'
import { StateProps, DispatchProps } from '../containers/StartScreen'
import { NavigationInjectedProps } from 'react-navigation'

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
})

export interface Props extends StateProps, DispatchProps, NavigationInjectedProps {}

export default class StartScreen extends React.Component<Props, object> {
  private renderWorld = (): string => 'World'

  public render(): JSX.Element {
    const { counter, addCounter, navigation } = this.props

    return (
      <View style={styles.container}>
        <Text>Hello {this.renderWorld()}!</Text>
        <Text>Clicks: {counter}</Text>
        <Button onPress={addCounter} title="Click me" color="#841584" />
        <Button title="Go to Second Screen" onPress={() => navigation.navigate('SecondScreen')} />
      </View>
    )
  }
}
