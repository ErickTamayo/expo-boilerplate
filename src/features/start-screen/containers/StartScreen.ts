import { connect } from 'react-redux'
import { Dispatch } from 'redux'
import { State } from '../records/StartScreenState'
import StartScreen from '../components/StartScreen'
import { addCounter, AddCounter } from '../actions'

export interface StateProps {
  counter: number
}

export interface DispatchProps {
  addCounter: () => AddCounter
}

export const mapStateToProps = ({ startScreen }: { startScreen: State }): StateProps => ({
  counter: startScreen.counter,
})

export const mapDispatchToProps = (dispatch: Dispatch): DispatchProps => ({
  addCounter: () => dispatch(addCounter()),
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(StartScreen)
