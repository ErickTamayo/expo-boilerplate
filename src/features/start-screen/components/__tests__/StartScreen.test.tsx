import 'react-native'
import * as React from 'react'
import { shallow } from 'enzyme'
import StartScreen, { Props } from '../StartScreen'
import { AddCounter } from '../../actions'
import { NavigationScreenProp, NavigationRoute, NavigationParams } from 'react-navigation'

describe('<StartScreen />', () => {
  it('renders correctly', () => {
    const props: Props = {
      counter: 0,
      addCounter: () => ({} as AddCounter),
      navigation: {} as NavigationScreenProp<NavigationRoute<NavigationParams>, NavigationParams>,
    }
    const wrapper = shallow(<StartScreen {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
})
