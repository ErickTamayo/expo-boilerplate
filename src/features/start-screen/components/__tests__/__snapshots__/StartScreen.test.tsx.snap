// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StartScreen /> renders correctly 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Component>
    Hello 
    World
    !
  </Component>
  <Component>
    Clicks: 
    0
  </Component>
  <Button
    color="#841584"
    onPress={[Function]}
    title="Click me"
  />
  <Button
    onPress={[Function]}
    title="Go to Second Screen"
  />
</Component>
`;
